src/main/java/de/codeschluss/wooportal/server/components/activity/ActivityController.java

```diff
@@ -631,5 +652,15 @@ private void validateImages(List<ImageEntity> images) {
     } catch (NotFoundException e) {
       throw new BadParamsException("Given Activity does not exist");
     }
-  }
+   
+    } 
+
+    @GetMapping("/activities/{activityId}/iCal")
+    public ResponseEntity<String> generateCalenderFile(@PathVariable String activityId)
+
+
+    {
+      return ok(service.generateCalenderFile(activityId));
```

Comment:
Set "content disposition" header so that the browser can do automatic file download of the generated file